### PR TITLE
Code Clean Up for Platfrom Controller 

### PR DESCRIPTION
**Pull request by SuyogShinde942**
This PR resolves issues #1371 and #1370:

Cleanup Unused Resource Policy Controller (#1371):

Removed the unused resourcepolicycontroller.go file.
Deleted commented-out references to resourcepolicycontroller in main.go, as it was never utilized.
Enhancements in Resource Composition Handling (#1370):

Improved PlatformController by ensuring the handleCRD method registers the CRD defined in the ResourceComposition spec.

### CODE REVIEW

File: platform-operator/platformcontroller.go [hunk 1]
```diff
@@ -566,15 +566,15 @@ func createResourcePolicy(resPolicySpec interface{}, namespace string) {
 	}
 }
 
-func deleteResourcePolicy(resPolicySpec interface{}, namespace string) {	
+func deleteResourcePolicy(resPolicySpec interface{}, namespace string) {
 	fmt.Println("Inside deleteResourcePolicy.")
 	resPolicyObject := resPolicySpec.(platformworkflowv1alpha1.ResourcePolicy)
 	inputResPolicyName := resPolicyObject.ObjectMeta.Name
 	/*namespace := resPolicyObject.ObjectMeta.Namespace
```
Comment (devdattakulkarni) [inline platform-operator/platformcontroller.go:573]: We can delete these commented out lines as well.

File: platform-operator/platformcontroller.go [hunk 2]
```diff
@@ -811,17 +729,17 @@ func handleCRD(rescomposition, kind, version, group, plural, action, namespace,
 
 	if !crdPresent {
 		if action == "create" {
-		/*	_, err1 := crdClient.CustomResourceDefinitions().Create(context.Background(), crd, metav1.CreateOptions{})
-			if err1 != nil {
-				panic(err1.Error())
+			/*	_, err1 := crdClient.CustomResourceDefinitions().Create(context.Background(), crd, metav1.CreateOptions{})
```
Comment (devdattakulkarni) [inline platform-operator/platformcontroller.go:732]: We can delete these commented out lines (732-736).

**Comment by devdattakulkarni** [comment]
@SuyogShinde942 Mostly looks good. There are some commented lines that we can delete as well.

**Comment by SuyogShinde942** [comment]
I have removed the commented and some debugging code

File: platform-operator/main.go [hunk 3]
```diff
@@ -58,10 +57,10 @@ func main() {
 	}()
 
 	/*
```
Comment (devdattakulkarni) [inline platform-operator/main.go:59]: We can also remove the comment strings (/* */)

File: platform-operator/platformcontroller.go [hunk 4]
```diff
@@ -603,57 +594,50 @@ func deleteResourcePolicy(resPolicySpec interface{}, namespace string) {
 
 func flatten(yaml_contents map[string]interface{}, types_dict map[string]apiextensionsv1beta1.JSONSchemaProps) map[string]apiextensionsv1beta1.JSONSchemaProps {
```
Comment (devdattakulkarni) [inline platform-operator/platformcontroller.go:591]: Can you verify if flatten method is being referred to anywhere?
If not, we can delete it.

**Comment by devdattakulkarni** [comment]
@SuyogShinde942 I have added some more comments.

File: platform-operator/platformcontroller.go (hunk 4 above)
Comment (SuyogShinde942) [reply platform-operator/platformcontroller.go:591]: sure

Comment (SuyogShinde942) [reply platform-operator/platformcontroller.go:591]: We are using it in the getChartValueTypes function

Comment (devdattakulkarni) [reply platform-operator/platformcontroller.go:591]: But getChartValueTypes() is part of the commented out code that you have removed.
So I think we can safely remove flatten().

File: platform-operator/platformcontroller.go [hunk 5]
```diff
@@ -603,57 +590,50 @@ func deleteResourcePolicy(resPolicySpec interface{}, namespace string) {
...
 
 			types_dict[key] = jsonSchemaInner
 		}
 
 	}
-	//fmt.Println(types_dict)
 	return types_dict
 }
 
 func getChartValueTypes(data []byte) map[string]apiextensionsv1beta1.JSONSchemaProps {
```
Comment (devdattakulkarni) [inline platform-operator/platformcontroller.go:633]: I think we can get rid of this method as well.

**Comment by devdattakulkarni** [comment]
@SuyogShinde942 I have added couple of more comments.

**Pull request closed without merging**